allennlp/training/tensorboard_writer.py

```diff
@@ -61,210 +64,94 @@ def __init__(
         self,
         serialization_dir: Optional[str] = None,
         summary_interval: int = 100,
-        histogram_interval: int = None,
+        distribution_interval: Optional[int] = None,
+        histogram_interval: Optional[int] = None,
         batch_size_interval: Optional[int] = None,
         should_log_parameter_statistics: bool = True,
         should_log_learning_rate: bool = False,
+        should_log_inputs: bool = False,
         get_batch_num_total: Callable[[], int] = None,
     ) -> None:
-        if serialization_dir is not None:
+
+        if distribution_interval is not None and histogram_interval is not None:
+            logger.warning(
+                f"`histogram_interval`=`{histogram_interval}` will be ignored "
+                f"in favor of `distribution_interval`=`{distribution_interval}`"
+            )
+
+        if distribution_interval is None:
+            distribution_interval = histogram_interval
```

Comment:
I think we should issue a deprecation warning instead. And instead of listing `histogram_interval` as an actual parameter, we could just check for it in `**kwargs`.

Like what we did here: https://github.com/allenai/allennlp/blob/1cff6ad97fe8304d03923ddfaedb5e05bb67bee5/allennlp/data/dataset_readers/conll2003.py#L99-L104